test/jdk/tools/sincechecker/SinceChecker.java

```diff
@@ -0,0 +1,769 @@
+                    + mappedVersion + " while the `@since` in the source code is= " + sinceVersion);
+            return;
+        }
+        if (Integer.parseInt(sinceVersion) < 9) {
+            sinceVersion = "9";
+        }
+        if (!sinceVersion.equals(mappedVersion)) {
+            String message = getWrongSinceMessage(sinceVersion, mappedVersion, id);
+            wrongTagsList.add(message);
+        }
+    }
+
+    private static String getWrongSinceMessage(String sinceVersion, String mappedVersion, String elementSimpleName) {
+        String message;
+        if (mappedVersion.equals("9")) {
+            message = "For " + elementSimpleName +
+                    " Wrong `@since` version " + sinceVersion + " But the element exists before JDK 10\n";
+        } else {
+            message = "For " + elementSimpleName +
+                    " Wrong `@since` version is " + sinceVersion + " instead of " + mappedVersion + "\n";
+        }
+        return message;
+    }
+
+    private static String getElementName(Element owner, Element element, Types types) {
+        String prefix = "";
+        String suffix = "";
+        ElementKind kind = element.getKind();
+        if (kind.isField()) {
+            TypeElement te = (TypeElement) owner;
+            prefix = "field";
+            suffix = ": " + te.getQualifiedName() + ":" + element.getSimpleName();
+        } else if (kind == ElementKind.METHOD || kind == ElementKind.CONSTRUCTOR) {
+            prefix = "method";
+            TypeElement te = (TypeElement) owner;
+            ExecutableElement executableElement = (ExecutableElement) element;
+            String returnType = types.erasure(executableElement.getReturnType()).toString();
+            String methodName = executableElement.getSimpleName().toString();
+            String descriptor = executableElement.getParameters().stream()
+                    .map(p -> types.erasure(p.asType()).toString())
+                    .collect(Collectors.joining(",", "(", ")"));
+            suffix = ": " + returnType + " " + te.getQualifiedName() + "." + methodName + descriptor;
+        } else if (kind.isDeclaredType()) {
+            if (kind.isClass()) {
+                prefix = "class";
+            } else if (kind.isInterface()) {
+                prefix = "interface";
+            }
+            suffix = ": " + ((TypeElement) element).getQualifiedName();
+        } else if (kind == ElementKind.PACKAGE) {
+            prefix = "package";
+            suffix = ": " + ((PackageElement) element).getQualifiedName();
+        } else if (kind == ElementKind.MODULE) {
+            prefix = "module";
+            suffix = ": " + ((ModuleElement) element).getQualifiedName();
+        }
+        return prefix + suffix;
+    }
+
+    public static class IntroducedIn {
+        public String introducedPreview;
+        public String introducedStable;
+    }
```

Comment:
Suggest move this nearer the map at the top of the class -- either move this class up, or those members down.

Reply:
Thanks, moved it up